sessionctx/variable/varsutil.go

```diff
@@ -156,6 +165,181 @@ func ValidateGetSystemVar(name string, isGlobal bool) error {
 	return nil
 }
 
+// ValidateSetSystemVar checks if system variable satisfies specific restriction.
+func ValidateSetSystemVar(name string, value string) (string, error, error) {
+	if strings.EqualFold(value, "DEFAULT") {
+		if val := GetSysVar(name); val != nil {
+			return val.Value, nil, nil
+		}
+		return value, nil, nil
```

Comment:
Any case cover this line?

Reply:
In `set_test.go`, line 93 - line 97 covers this:
```go
	tk.MustQuery(`select @@global.low_priority_updates;`).Check(testkit.Rows("0"))
	tk.MustExec(`set @@global.low_priority_updates="ON";`)
	tk.MustQuery(`select @@global.low_priority_updates;`).Check(testkit.Rows("1"))
	tk.MustExec(`set @@global.low_priority_updates=DEFAULT;`) // It will be set to compiled-in default value.
	tk.MustQuery(`select @@global.low_priority_updates;`).Check(testkit.Rows("0"))
	
```

Reply:
Yeah, this case covers line 172, but seems there is no case that covers line 174?

Reply:
@XuHuaiyu  the condition that we fail to `GetSysVar` should never happen. We'd better panic here ?